flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/WindowAggregateReduceFunctionsRule.scala

```diff
@@ -39,14 +40,26 @@ import scala.collection.JavaConversions._
 class WindowAggregateReduceFunctionsRule
   extends AggregateReduceFunctionsRule(
     operand(classOf[LogicalWindowAggregate], any()),
-    RelFactories.LOGICAL_BUILDER) {
+    RelBuilder.proto(
+      Contexts.of(
+        RelFactories.DEFAULT_STRUCT,
+        RelBuilder.Config.DEFAULT
+          .withPruneInputOfAggregate(false)))) {
```

Comment:
How about introducing Flink's own `RelFactories` to avoid someone forgotten to disable this?

Reply:
This is a `RelBuilder`, not `RelFactories`.